### PR TITLE
HDS-1591-replace-form-validation-library

### PR DESCRIPTION
**Pull request by mrTuomoK**
## Description
Replace old Formik library with more current React Hook Form library

## Related Issue
[HDS-1591](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1591)

## Motivation and Context
Formik is kind of obsolete nowadays so thus why this change.

## How Has This Been Tested?
- local machine, should match the old implementation

[HDS-1591]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/react/src/examples/forms/validationDynamic.stories.tsx [hunk 1]
```diff
@@ -1,192 +1,122 @@
...
-      if (invalidFields.length > 0) {
-        const firstFieldId = getFocusableFieldId(invalidFields[0]);
-        document.getElementById(firstFieldId).focus();
-      } else {
-        formik.submitForm();
+  React.useEffect(() => {
+    const invalidFields = Object.keys(errors);
+    // check for city and if it's the 1st -> focus
+    if (errors?.city) {
+      if (invalidFields[0] === 'city') {
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationDynamic.stories.tsx:67]:
```suggestion
    if (errors?.city && invalidFields[0] === 'city') {
```

File: packages/react/src/examples/forms/validationDynamic.stories.tsx [hunk 2]
```diff
@@ -1,192 +1,122 @@
...
+            element.focus();
+          }, 10);
+        }
       }
-    });
-  };
+      return;
+    }
+    if (errors?.permitEndDate) {
+      if (invalidFields[0] === 'permitEndDate') {
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationDynamic.stories.tsx:78]:
```suggestion
    if (errors?.permitEndDate && invalidFields[0] === 'permitEndDate') {
```

File: packages/react/src/examples/forms/validationHybrid.stories.tsx [hunk 3]
```diff
@@ -1,170 +1,112 @@
...
-    });
+  const handleChange = (name, value) => {
+    setValue(name, value, { shouldTouch: true, shouldValidate: true });
   };
 
+  React.useEffect(() => {
+    const invalidFields = Object.keys(errors);
+    // check for city and if it's the 1st -> focus
+    if (errors?.city) {
+      if (invalidFields[0] === 'city') {
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationHybrid.stories.tsx:88]:
```suggestion
    if (errors?.city && invalidFields[0] === 'city') {
```

File: packages/react/src/examples/forms/validationHybrid.stories.tsx [hunk 4]
```diff
@@ -1,170 +1,112 @@
...
+        if (element) {
+          setTimeout(() => {
+            element.focus();
+          }, 10);
+        }
+      }
+      return;
+    }
+    if (errors?.permitEndDate) {
+      if (invalidFields[0] === 'permitEndDate') {
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationHybrid.stories.tsx:99]:
```suggestion
    if (errors?.permitEndDate && invalidFields[0] === 'permitEndDate') {
```

File: packages/react/src/examples/forms/validationStatic.stories.tsx [hunk 5]
```diff
@@ -129,41 +67,58 @@ export const Static = () => {
...
+  const onSubmitHandler = (data) => {
+    console.log({ data });
+    reset();
   };
 
+  React.useEffect(() => {
+    const invalidFields = Object.keys(errors);
+    // check for city and if it's the 1st -> focus
+    if (errors?.city) {
+      if (invalidFields[0] === 'city') {
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationStatic.stories.tsx:79]:
```suggestion
    if (errors?.city && invalidFields[0] === 'city') {
```

File: packages/react/src/examples/forms/validationStatic.stories.tsx [hunk 6]
```diff
@@ -129,41 +67,58 @@ export const Static = () => {
...
+        if (element) {
+          setTimeout(() => {
+            element.focus();
+          }, 10);
+        }
+      }
+      return;
+    }
+    if (errors?.permitEndDate) {
+      if (invalidFields[0] === 'permitEndDate') {
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationStatic.stories.tsx:90]:
```suggestion
    if (errors?.permitEndDate && invalidFields[0] === 'permitEndDate') {
```

**Review by mikkojamG — CHANGES_REQUESTED** [review]

File: packages/react/src/examples/forms/validationDynamic.stories.tsx [hunk 7]
```diff
@@ -200,45 +130,42 @@ export const Dynamic = () => {
                 optionLabelField="label"
                 options={cities}
                 onChange={(selected: CityOptionType) => {
-                  formik.setFieldValue('city', selected ? selected.label : '');
+                  handleChange('city', selected ? selected.label : '');
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationDynamic.stories.tsx:133]: Looks like typing the value for city is broken for both dynamic and hybrid examples 🤔 Clicking on the list works.

form-validation

File: packages/react/src/examples/forms/validationStatic.stories.tsx [hunk 8]
```diff
@@ -172,28 +127,32 @@ export const Static = () => {
           <div className="hds-example-form__grid-6-6">
             <div className="hds-example-form__item">
               <TextInput
+                {...register('firstName')}
+                onChange={(e) => {
+                  handleChange('firstName', e.currentTarget.value);
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationStatic.stories.tsx:132]: Validation doesn't seem to be working for static examples 🤔 

form-validation-static

**Review by mikkojamG — CHANGES_REQUESTED** [review]

File: packages/react/src/examples/forms/validationDynamic.stories.tsx (hunk 7 above)
Comment (mrTuomoK) [reply packages/react/src/examples/forms/validationDynamic.stories.tsx:128]: Ah true, I might have broken these somewhere in the middle... used to work :D

File: packages/react/src/examples/forms/validationStatic.stories.tsx [hunk 9]
```diff
@@ -172,28 +122,32 @@ export const Static = () => {
...
+                errorText={errors.firstName?.message}
                 autoComplete="given-name"
                 required
               />
             </div>
             <div className="hds-example-form__item">
               <TextInput
+                {...register('lastName')}
+                onChange={(e) => {
+                  handleChange('firstName', e.currentTarget.value);
```
Comment (mikkojamG) [inline packages/react/src/examples/forms/validationStatic.stories.tsx:143]: Noticed small mistake here: pointing to `firstName` field 🙂

**Review by mikkojamG — CHANGES_REQUESTED** [review]

File: packages/react/src/examples/forms/validationStatic.stories.tsx (hunk 9 above)
Comment (mrTuomoK) [reply packages/react/src/examples/forms/validationStatic.stories.tsx:143]: Oh, good spottind :D too much copy & paste it seems :D :D

**Review by mikkojamG — APPROVED** [review]
LGTM 👍

**Comment by mrTuomoK** [comment]
Ah Ok I'll check if I can fix it by tomorrow.

**Review by harriplappalainen — APPROVED** [review]
Yes, this is brilliant 🚀

**Pull request merged**